### DISCUSSION TITLE
Could not find system Git

### DISCUSSION THREAD

**Discussion by AilsaWatson** [IBAHCM/RPiR#444] [Q&A for technical issues] [answered]
Hi @richardreeve @soniamitchell ,

I've been working on series03 without any luck in pushing anything to GitHub. I can't now seem to push packages (or projects) there either. 

I get this error message at this point: 
> gitcreds::gitcreds_set()
Error in throw(new_error("gitcreds_nogit_error")) : 
  Could not find system git

I have tried;

1.  uninstalling and reinstalling Git
2. In advanced system settings, adding 'C:\Program Files\Git\bin\git.exe' to the PATH (where 'C:\Program Files\Git\cmd' already was) & checking with command Prompt that Git is installed / it exists in the PATH. Here is a link to the Stackoverflow with these steps - https://stackoverflow.com/questions/26620312/git-installing-git-in-path-with-github-client-for-windows

I will keep trying but not sure what else I can do. 

Thank you for any help!

**Comment by richardreeve** [comment, marked as answer]
Hi @AilsaWatson - if you go to `Tools -> Global options` and choose the `git/svn` tab (it'll be something like that - I don't have a Windows machine), then you need to browse for the git executable - presumably in `C:\Program Files\Git\bin\git.exe`, assuming that file exists? You should then restart RStudio. See if that fixes it?

**Comment by alexpate30** [comment]
Hello, I have encountered the same problem. I have set the Git executable path in both "tools > project options > Git/SVN" and "tools > global options > Git/SVN" to be the git executable file (hidden away as part of GitHub desktop so the file path is a bit longer "C:Users/.../AppData/Local/GitHubDesktop/app-3.2.3/resources/app/git/cmd/git.exe").

I am using R 4.3.0. Rstudio 2023.03.0+386. Github desktop version 3.2.3. I have uninstalled and reinstalled all. Any ideas?

**Reply by richardreeve** [reply]
Apologies for the extreme delay - I don't monitor this repo when I'm not teaching. I hope you got it fixed. If not though, that doesn't look like a valid Windows path to be honest. Surely it should at least be `C:/User/...` or more likely `C:\Users\...`?

**Reply by alexpate30** [reply]
No problem. Issue was sorted by installing Git separately from GitHub desktop, and directing the path to this. The path above is the path to the Git executable that is installed with GitHub desktop, which seemed to be causing the issue.

**Comment by fdyulin** [comment]
first, open git CMD to see your username. It is MINGW64 for me.
Then, find C:\Program Files\Git\mingw64\bin\git.exe.
Finally,set this file as the git executable file in "tools > global options > Git/SVN". And restart your computer.
Problem solved.

**Reply by Veny-hkw** [reply]
please est-il possible d'avoir un exemple avec des images à l'appui car je ne comprends pas juste avec les mots.
